cmd/tpm-v2.c

```diff
@@ -8,11 +8,43 @@
 #include <command.h>
 #include <dm.h>
 #include <log.h>
+#include <malloc.h>
 #include <mapmem.h>
 #include <tpm-common.h>
 #include <tpm-v2.h>
 #include "tpm-user-utils.h"
 
+static int do_tpm2_get_random(struct cmd_tbl *cmdtp, int flag, int argc,
+			      char *const argv[])
+{
+	struct udevice *dev;
+	char *buffer;
+	u32 len;
+	int ret;
+
+	ret = get_tpm(&dev);
+	if (ret) {
+		printf("Cant get tpm\n");
+		return ret;
+	}
+
+	if (argc != 2)
+		return CMD_RET_USAGE;
+
+	len = simple_strtoul(argv[1], NULL, 10);
+	buffer = calloc(1, len);
```

Comment:
tpm2_get_random() doesn't use data from the buffer and can't succeed until filling the buffer completely. Probably, we might use malloc() here instead.

Reply:
dont follow. what is wrong with using calloc?

Reply:
It just needs more operations to fill the memory with 0.

Reply:
why

Reply:
If some latest fetched bytes before failing/abortion are 0, we'll never know how many bytes are filled looking into the buffer.

Reply:
of course, however you still have a chance of getting meaningful info for debugging...doing what you propose has 0 chance. so I wont replace calloc :)

Reply:
Ah, now I see, thanks. Make sense.

I would rather convert the count parameter of tpm2_get_random() (and other funcs) into *count so that in case of an error a caller always knows how many bytes remain to be done.
I'll ask Simon what does he think about such changes in the API.